### PR TITLE
chore: move CloseButton to the ui library

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
also cut the dependency to tinro and parent element need to handle the dispatch

### What issues does this PR fix or reference?

need it for https://github.com/containers/podman-desktop/issues/6920

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/ui/src/lib/button/CloseButton.svelte [hunk 1]
```diff
@@ -2,26 +2,21 @@
 import { faTimes } from '@fortawesome/free-solid-svg-icons';
 import { createEventDispatcher } from 'svelte';
 import Fa from 'svelte-fa';
-import { router } from 'tinro';
 
-export let href: string | undefined = undefined;
+export let href: string | undefined = 'undefined';
```
Comment (lstocchi) [inline packages/ui/src/lib/button/CloseButton.svelte]: I wonder why we are keeping the href here. Before there was a different behavior based on the value href had. If it was undefined a click was dispatched, if it was defined it worked like a link. 
In the new way we always trigger the click. Can we remove this href?

Comment (benoitf) [reply packages/ui/src/lib/button/CloseButton.svelte:6]: ah it's a leftover from a test, it should not be 'undefined'

Comment (benoitf) [reply packages/ui/src/lib/button/CloseButton.svelte]: so now I see it's a suggestion

the idea is that if you expect an href you'll get a string, else you receive undefined

Comment (benoitf) [reply packages/ui/src/lib/button/CloseButton.svelte]: you always get a click event, then it's either containing a value or undefined

Comment (benoitf) [reply packages/ui/src/lib/button/CloseButton.svelte]: I kept the href so when we get the click event we receive the value, the won't need to keep it on the parent component as an attribute

Comment (benoitf) [reply packages/ui/src/lib/button/CloseButton.svelte]: but we could replace `href` by `clickDetail` or `detail`

Comment (lstocchi) [reply packages/ui/src/lib/button/CloseButton.svelte]: it looks to me too complex tbh, we could just create the close button component with a callback that gets called on the click.
If you want to keep the current logic, yes maybe renaming would reduce the confusion imo

Comment (benoitf) [reply packages/ui/src/lib/button/CloseButton.svelte]: > it looks to me too complex tbh, we could just create the close button component with a callback that gets called on the click.

it's what it does, it call the optional on:click method

but I can remove the field

waiting for other reviews to decide

Comment (lstocchi) [reply packages/ui/src/lib/button/CloseButton.svelte]: I was referring to a simplification like the following (so we remove the href, dispatch and click() from the CloseButton) 
```
export let onClose: () => {};

on:click="{onClose}"
```

then from the caller
```
<CloseButton
   onClose=() => { .... }
```

**Comment by benoitf** [comment]
so I used the same logic than the <Link component, we don't provide anymore the href field and we have `on:click` event (as any button)

File: packages/ui/src/lib/button/CloseButton.svelte (hunk 1 above)
Comment (benoitf) [reply packages/ui/src/lib/button/CloseButton.svelte]: I changed the implementation to match Link implementation

**Review by lstocchi — APPROVED** [review]
LGTM

**Review by deboer-tim — APPROVED** [review]

**Pull request merged**